### ISSUE TITLE
Change JSON encoding of public keys

### ISSUE THREAD

**Issue by heartsucker** [heartsucker/rust-tuf#80]
Currently uses:

```json
{
  "key_type": "ed25519 | rsa",
  "key_value": {
    "public": "<PEM> | <hexlower>"
  }
}
```

If we go with PKCS#8 everywhere, then the above can be replaced with a PEM string since all of the information is contained therein, and we're not faffing about with having to keep track of it with a `KeyFormat` enum for deterministic serialization later.

Related: #79 #78

**Comment by heartsucker** [comment]
Related #66

**Comment by heartsucker** [comment]
Closed by 978d7ce
